packages/contractkit/src/wallets/ledger-wallet.ts

```diff
@@ -0,0 +1,222 @@
+import { isHexString, normalizeAddressWith0x, trimLeading0x } from '@celo/utils/lib/address'
+import { TransportError, TransportStatusError } from '@ledgerhq/errors'
+import Ledger from '@ledgerhq/hw-app-eth'
+import { byContractAddress } from '@ledgerhq/hw-app-eth/erc20'
+import debugFactory from 'debug'
+import * as ethUtil from 'ethereumjs-util'
+import { EncodedTransaction, Tx } from 'web3-core'
+import { Address } from '../base'
+import { EIP712TypedData, generateTypedDataHash } from '../utils/sign-typed-data-utils'
+import {
+  chainIdTransformationForSigning,
+  encodeTransaction,
+  rlpEncodedTx,
+  signatureFormatter,
+} from '../utils/signing-utils'
+import { Wallet } from './wallet'
+
+export const CELO_BASE_DERIVATION_PATH = "44'/52752'/0'/0"
+const ADDRESS_QTY = 5
+
+export async function newLedgerWalletWithSetup(
+  transport: any,
```

Comment:
I think you can make this the `Transport` type: `import Transport from '@ledgerhq/hw-transport'`